tensorbay/utility/file.py

```diff
@@ -19,6 +19,55 @@
 from tensorbay.utility.repr import ReprMixin
 
 
+class Url:
+    """Url is a class used to get and update the url.
+
+    Arguments:
+        url: The url.
+        updater: A function used to update the url.
+
+    """
+
+    def __init__(self, url: str, updater: Callable[..., Optional[str]]) -> None:
+        self._url = url
```

Comment:
`self._url` seems useless, the cached url can be put in to the closure `_getter`